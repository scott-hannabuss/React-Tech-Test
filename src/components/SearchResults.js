import React from "react";
import "../styles/SearchResults.css";
import PropTypes from "prop-types";


const SearchResults = ({results}) => {
    if (!results.length) {
        return <p className="search-info">Please enter a relevant search term to return image results from Nasa's API</p>
      }
      else {
        return (
    <>
      {results.map((image) => (
        <div className="image-container">
          <img className="card-image" src={image} alt="spaceImage" />
          </div>
        ))}
      </>
  );
};
}

SearchResults.propTypes = {
    results: PropTypes.array.isRequired,
  };


export default SearchResults;