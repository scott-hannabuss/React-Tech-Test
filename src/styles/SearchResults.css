.search-results {
  column-count: 5;
  column-gap: 15px;
  margin: 10px;
}

.image-container {
  display: inline-block;
  width: 100%;
  margin-top: 5px;
}

.image-container img {
  display: block;
  width: 100%;
  border-radius: 25px;
}

.search-results > p {
  column-span: all;
}

.search-info {
  font-family: "Trebuchet MS", sans-serif;
  text-align: center;
}

@media only screen and (max-width: 750px) {
  .search-results {
    column-count: 3;
    column-gap: 5px;
    margin: 25px;
  }

  .image-container {
    margin-top: 1px;
  }

  .image-container img {
    border-radius: 5px;
  }
}
