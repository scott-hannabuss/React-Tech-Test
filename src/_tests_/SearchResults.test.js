import { render } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("<SearchResults />", () => {
  const images = [
    "https://images-assets.nasa.gov/image/PIA08946/PIA08946~thumb.jpg",
    "https://images-assets.nasa.gov/image/PIA09000/PIA09000~thumb.jpg",
  ];

  it("renders and matches snapshot", () => {
    const { asFragment } = render(<SearchResults results={images} />);
    expect(asFragment()).toMatchSnapshot();
  });
});