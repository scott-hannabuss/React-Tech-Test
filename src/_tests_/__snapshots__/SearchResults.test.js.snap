// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResults /> renders and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="image-container"
  >
    <img
      alt="spaceImage"
      class="card-image"
      src="https://images-assets.nasa.gov/image/PIA08946/PIA08946~thumb.jpg"
    />
  </div>
  <div
    class="image-container"
  >
    <img
      alt="spaceImage"
      class="card-image"
      src="https://images-assets.nasa.gov/image/PIA09000/PIA09000~thumb.jpg"
    />
  </div>
</DocumentFragment>
`;
